spec/integration_spec.rb

```diff
@@ -109,6 +109,32 @@ def pptx(pptx_name)
     end
   end
 
+  context "with a slide with text" do
+    let(:theme) { OpenXml::Pptx::Parts::Theme.new }
+    let(:master) { OpenXml::Pptx::Parts::SlideMaster.new(theme) }
+    let(:layout) { OpenXml::Pptx::Parts::SlideLayout.new(master) }
+    let(:slide) { OpenXml::Pptx::Parts::Slide.new(layout) }
+
+    before do
+      slide.add_text("Hello World")
```

Comment:
> Is it okay to extract the shape and complexity of the xml away from the user? I am not sure at this time.

I think there are three moves that we can **always** make to remove complexity from the user:

 1. Hiding values that are used only internally (like object IDs and mime types)
 2. Giving attributes sensible defaults
 3. Automatically constructing required dependencies — `<p:sp>` **_always_** contains both a `<p:nvSpPr>` and a `<p:spPr>` (XSD below), so `Shape#initialize` should construct both `ShapeNonVisual` and `ShapeProperties` and expose them via attrs for potential modification

There are so many ways we could construct a pleasant DSL on top of all this complexity, but I want to defer decisions of that sort until we know more. In the meantime, focusing on these three ways to remove complexity will both go a long way and be a safe bet.

* * * 

It doesn't seem onerous to me that the user should have to construct a `Shape`, `TextBox`, `Paragraph`, and `Run` to add text to a slide because — as soon as their use-case is more than a simple "Hello World" example — they'll want to:

 - set the position and dimensions of the `Shape`
 - put something else in the shape rather than a `TextBox` — like an `Image`
 - manipulate the `Paragraph`'s properties
 - add more than one `Run` to the paragraph
 - etc


*  *  *

```
<xsd:complexType name="CT_Shape">
  <xsd:sequence>
    <xsd:element name="nvSpPr" type="CT_ShapeNonVisual" minOccurs="1" maxOccurs="1"/>
    <xsd:element name="spPr" type="a:CT_ShapeProperties" minOccurs="1" maxOccurs="1"/>
    <xsd:element name="style" type="a:CT_ShapeStyle" minOccurs="0" maxOccurs="1"/>
    <xsd:element name="txBody" type="a:CT_TextBody" minOccurs="0" maxOccurs="1"/>
  </xsd:sequence>
  ...
```